botfront/imports/api/nlu_model/nlu_model.methods.js

```diff
@@ -136,6 +136,37 @@ Meteor.methods({
             { $pull: { 'training_data.fuzzy_gazette': { _id: itemId } } },
         );
     },
+
+    'nlu.upsertRegexFeature'(modelId, item) {
+        check(modelId, String);
+        check(item, Object);
+        if (item._id) {
+            return NLUModels.update(
+                { _id: modelId, 'training_data.regex_features._id': item._id },
+                {
+                    $set: {
+                        'training_data.regex_features.$': item,
+                    },
+                },
+            );
+        }
+
+        const regexFeature = { ...item, _id: uuidv4() };
+        return NLUModels.update(
+            { _id: modelId },
+            { $push: { 'training_data.regex_features': regexFeature } },
+        );
+    },
```

Comment:
Suggestion : It could be great to have little bit of validation here, like trying to create a regex object with the regex string, and if it's fails displaying an error on the ui  as an alert ? since regexes can get big and cryptic someone could easily do a typo or remove a character that would break the regex.